src/common/runtime.rs

```diff
@@ -0,0 +1,275 @@
+use crate::{DecoModel, Depth, Gas, Minutes};
+
+use super::{AscentRatePerMinute, DecoModelConfig, DiveState, MbarPressure};
+
+const DEFAULT_ASCENT_RATE: AscentRatePerMinute = 9.;
+// todo move to model config
+const DEFAULT_CEILING_WINDOW: Depth = 3.;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum DecoAction {
+    AscentToCeil,
+    AscentToGasSwitchDepth,
+    Stop,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DecoStageType {
+    Ascent,
+    DecoStop,
+    GasSwitch
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct DecoStage {
+    pub stage_type: DecoStageType,
+    pub start_depth: Depth,
+    pub end_depth: Depth,
+    pub duration: Minutes,
+    pub gas: Gas,
+}
+
+#[derive(Clone, Debug)]
+pub struct DecoRuntime {
+    pub deco_stages: Vec<DecoStage>,
+    pub tts: Minutes,
+}
+
+impl DecoRuntime {
+    pub fn new() -> Self {
+        Self {
+            deco_stages: vec![],
+            tts: 0,
+        }
+    }
+
+    pub fn calc(&mut self, mut sim_model: impl DecoModel, gas_mixes: Vec<Gas>) -> Self {
+        // loop until no deco action
+        loop {
```

Comment:
Ty weź to rozbij bo to nieczytelne jest totalnie

Reply:
Byś był wtedy na callu to byś zrozumial ze tu troszke piramida dziwna :D